### PR TITLE
Add README.md

### PR DESCRIPTION
**Pull request by daeyeon**
IoT.js-Modules-DCO-1.0-Signed-off-by: daeyeon.jeong@samsung.com

### CODE REVIEW

**Comment by rzr** [comment]
my remarks got lost,
I am just suggesting that hyphen can be used in package name, there are many node packages that are using it and can me "imported" without renaming.

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,191 @@
+Copyright 2018-present Samsung Electronics Co., Ltd. and other contributors
```
Comment (rzr) [inline LICENSE]: Maybe a copyright file is better to mention those line
and keep LICENCE a vanilla copy of Apache-2.0

I know that some github's bots are scanning this and can be fooled if modified.

File: README.md [hunk 2]
```diff
@@ -1 +1,47 @@
...
+
+## Introduction
+
+IoT.js Modules is a collection of packages which works with [IoT.js](https://github.com/Samsung/iotjs), its mother project. The purpose of this "central community repository" is to provide a place everyone can freely share and borrow the packages. It includes not only IoT.js compatible modules but also tools and demo projects which can get inspiration from.
+
+## Ground rules
+As of now, we set out to use git submodule to avoid raw duplication of each package. Tracking history and its integrity are guaranteed by git. The minimum requirements for the package are as follows:
+- Basically, each package should be given as a standalone gitmodule and be added as a submodule of this repository.
+- Each package should contain a descriptor, `package.json`, which includes `name`, `version` and `iotjs`.
+- Each package should not use IoT.js core module names such as `fs`, `http`, `console`, and so on. It's encouraged to align with the conventions of `Node.JS` as much as possible.
```
Comment (rzr) [inline README.md]: Note for later, I tend to add the -lite suffix and make it working also on node.

So for pending http "super module" we could rename it as http-lite that would base on IoT'js http (raw) module and eventually node one too.

**Comment by daeyeon** [comment]
@rzr Right. Updated. For rapid-initial-setup for this repository, I tried to take remarks only which we've used and discussed so far.

File: README.md [hunk 3]
```diff
@@ -1 +1,47 @@
...
+{
+  "name": "", // all lowercase. one word, no spaces. dashes and underscores allowed.
+  "version": "", // in the form of x.x.x.
+
+  .....
+
+  "iotjs": {
+    "version": ">=1.0"
+  }
+
```
Comment (rzr) [inline README.md]: see my previous comment about embedding into  "engines": scope...
but it's maybe be not justified, I noticed that node-red is doing this at the same level, so I am fine with it.

File: README.md [hunk 4]
```diff
@@ -1 +1,47 @@
...
+For further information, please refer to [package.json](https://docs.npmjs.com/files/package.json).
+
+### Adding a submodule
+```
+git submodule add https://github.com/{org}/{project}
+git commit -sam 'Import {module} to iotjs modules'
+```
+
+### Including Developer's Certificate of Origin ("DCO")
+
```
Comment (rzr) [inline README.md]: maybe can be moved upper about iotjs conventions

File: README.md [hunk 5]
```diff
@@ -1 +1,47 @@
...
+It's required to put a line at the end of your commit message which certifies who is the author. It aims to improve tracking of who did what, especially with patches. You should use your real name and email address in the format below:
+
+```
+IoT.js-Modules-DCO-1.0-Signed-off-by: Random Developer random@developer.example.org
+```
+## License
+IoT.js Modules is Open Source software under the [Apache 2.0 license](LICENSE).
```
Comment (rzr) [inline README.md]: Respectively imported modules are under compatibles licences,
as long as they are allowing to be imported as git modules.

File: README.md [hunk 6]
```diff
@@ -1 +1,47 @@
...
+- Each package should contain a descriptor, `package.json`, which includes `name`, `version` and `iotjs`.
+- Each package should not use IoT.js core module names such as `fs`, `http`, `console`, and so on. It's encouraged to align with the conventions of `Node.JS` as much as possible.
+- Other rules unmentioned follow [IoT.js Developer's Guide](https://github.com/Samsung/iotjs/wiki/Developer's-Guide) — e.g., One commit per one pull request, Signed-Off-By Process, Commit message convention and so on.
+
+## Examples
+### Adding a custom property for IoT.js to `package.json`.
+```js
+// package.json
+{
+  "name": "", // all lowercase. one word, no spaces. dashes and underscores allowed.
```
Comment (rzr) [inline README.md]: I think names can be relaxed to follow NPM rules

Comment (daeyeon) [reply README.md]: Sure. I didn't intend that. https://docs.npmjs.com/getting-started/using-a-package.json#requirements

File: README.md (hunk 3 above)
Comment (daeyeon) [reply README.md]: Can you share your suggestion a bit more? Is it like the following?

``{ "engines": {"iotjs":">=1.0"} }``

File: README.md (hunk 6 above)
Comment (daeyeon) [reply README.md]: Sure. I didn't intend any difference. https://docs.npmjs.com/getting-started/using-a-package.json#requirements

File: README.md (hunk 4 above)
Comment (daeyeon) [reply README.md]: It's unclear for me where exactly you prefer to.

File: README.md (hunk 3 above)
Comment (rzr) [reply README.md]: Likely yes but your readme can be merged as a base to start then I can submit PRs where we can debate on specific questions.

My idea was how to can we mention iotjs's dependencies if any

check this more relevant example:

```
{ "engines": 
   { "iotjs": { "version": >=1.0" ,
                  "dependencies":" { 
                          "mastodon-lite": "*", //package published on npm.org compatible with iotjs
                          "bmp085":  "https://github.com/TizenTeam/bmp085" // package forked for iotjs
                  }
   { "zephyr.js": (...) }
  } 
}
```

File: README.md (hunk 6 above)
Comment (rzr) [reply README.md]: sorry I missread (dashes and minus are same), so that's perfect,

File: README.md (hunk 5 above)
Comment (rzr) [reply README.md]: ok

**Comment by rzr** [comment]
ok to merge and improve later

File: README.md (hunk 4 above)
Comment (rzr) [reply README.md]: Maybe after line 12:
```
+- Other rules unmentioned follow [IoT.js Developer's Guide](https://github.com/Samsung/iotjs/wiki/Developer's-Guide) — e.g., One commit per one pull request, Signed-Off-By Process, Commit message convention and so on.
```

Not a big deal :)

File: LICENSE (hunk 1 above)
Comment (daeyeon) [reply LICENSE]: Out of the scope of this commit. Worth considering if the problem really happens.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**